src/features/roadmap/components/milestone-card.tsx

```
@@ -0,0 +1,95 @@
+    const handleStatusChange = async (status: "planned" | "in_progress" | "completed" | "archived") => {
+        try { await updateMilestone({ milestoneId: milestone._id, status }); toast.success(`Milestone marked as ${STATUS_CONFIG[status].label}`); }
+        catch { toast.error("Failed to update milestone"); }
+    };
+
+    const handleDelete = async () => {
+        try { await removeMilestone({ milestoneId: milestone._id }); toast.success("Milestone deleted"); }
+        catch { toast.error("Failed to delete milestone"); }
+    };
+
+    return (
+        <div className={cn("group rounded-lg border bg-white shadow-sm hover:shadow-md transition-shadow cursor-pointer overflow-hidden", milestone.status === "archived" && "opacity-60")} onClick={onClick}>
```

Biome (2.4.15)

[error] Static Elements should not be interactive.

(lint/a11y/noStaticElementInteractions)